src/transformers/modeling_tf_utils.py

```diff
@@ -3132,6 +3132,10 @@ def __init__(self, vocab_size: int, hidden_size: int, initializer_range: Optiona
         self.vocab_size = vocab_size
         self.hidden_size = hidden_size
         self.initializer_range = hidden_size**-0.5 if initializer_range is None else initializer_range
+        warnings.warn(
+            "TFSharedEmbeddings are scheduled for deletion, use `tf.keras.layers.Embedding` instead.",
```

Comment:
Could you put in a specific version in the warning? Typically >= 2 versions from the release this will be in